src/main/java/net/sf/jabref/gui/journals/ManageJournalAbbreviationsView.java

```diff
@@ -0,0 +1,546 @@
+                progressIndicator.setVisible(true);
+                viewModel.createFileObjects();
+                journalFilesBox.getSelectionModel().selectLast();
+                viewModel.addBuiltInLists();
+                loadingLabel.setVisible(false);
+                progressIndicator.setVisible(false);
+            }
+        }).start();
+    }
+
+    private void setButtonStyles() {
+        journalFilesBox.setPromptText(Localization.lang("No abbreviation files loaded"));
+        Text addJournalFileButtonGraphic = new Text(IconTheme.JabRefIcon.OPEN.getCode());
+        addJournalFileButtonGraphic.getStyleClass().add("icon");
+        addJournalFileButton.setGraphic(addJournalFileButtonGraphic);
+        Text addNewJournalFileButtonGraphic = new Text(IconTheme.JabRefIcon.NEW.getCode());
+        addNewJournalFileButtonGraphic.getStyleClass().add("icon");
+        addNewJournalFileButton.setGraphic(addNewJournalFileButtonGraphic);
+        Text removeJournalAbbreviationsButtonGraphic = new Text(IconTheme.JabRefIcon.CLOSE.getCode());
+        removeJournalAbbreviationsButtonGraphic.getStyleClass().add("icon");
+        removeJournalAbbreviationsButton.setGraphic(removeJournalAbbreviationsButtonGraphic);
+        ButtonBar.setButtonData(progressIndicator, ButtonData.LEFT);
+        ButtonBar.setButtonData(loadingLabel, ButtonData.LEFT);
+        ButtonBar.setButtonUniformSize(progressIndicator, false);
+        ButtonBar.setButtonUniformSize(loadingLabel, false);
+    }
+
+    private void setUpTable() {
+        journalAbbreviationsTable.setOnKeyPressed(event -> {
+            if ((event.getCode() == KeyCode.DELETE) && isEditableAndRemovable.get()) {
+                if ((viewModel.currentAbbreviationProperty().get() != null)
+                        && !viewModel.currentAbbreviationProperty().get().isPseudoAbbreviation()) {
+                    viewModel.deleteAbbreviation();
+                }
+            }
+        });
+        journalTableNameColumn.setCellValueFactory(cellData -> cellData.getValue().nameProperty());
+        journalTableNameColumn.setCellFactory(cell -> new JournalAbbreviationsNameTableEditingCell());
+        journalTableAbbreviationColumn.setCellValueFactory(cellData -> cellData.getValue().abbreviationProperty());
+        journalTableAbbreviationColumn.setCellFactory(cell -> new JournalAbbreviationsAbbreviationTableEditingCell());
+        journalTableEditColumn.setCellValueFactory(cellData -> cellData.getValue().isPseudoAbbreviationProperty());
+        journalTableDeleteColumn.setCellValueFactory(cellData -> cellData.getValue().isPseudoAbbreviationProperty());
+        journalTableEditColumn.setCellFactory(column -> new TableCell<AbbreviationViewModel, Boolean>() {
+
+            @Override
+            protected void updateItem(Boolean isPseudoAbbreviation, boolean isEmpty) {
+                super.updateItem(isPseudoAbbreviation, isEmpty);
+                if (isPseudoAbbreviation != null) {
```

Comment:
Yes it is necessary because we have to call `setGraphic(null)` if the item to update is null. Otherwise we get some weird graphic bugs when switching from a custom list to a built in list since the graphics don't get reset.

![journalabb](https://cloud.githubusercontent.com/assets/15340757/17476618/31371e1e-5d61-11e6-92d3-621bf8192548.PNG)